mu-plugins/central-hub/tests/phpunit/integration/config-store/loadConfigFromFilesystem.php

```diff
@@ -0,0 +1,110 @@
+ * @since      1.3.0
+ * @author     Robert A. Gadon
+ * @link       https://github.com/rgadon107/cornerstone
+ * @license    GNU General Public License 2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Integration\ConfigStore;
+
+use Brain\Monkey;
+use function KnowTheCode\ConfigStore\_the_store;
+use function KnowTheCode\ConfigStore\loadConfigFromFilesystem;
+use spiralWebDb\Cornerstone\Tests\Integration\Test_Case;
+
+
+/**
+ * Class Tests_LoadConfigFromFilesystem
+ *
+ * @package spiralWebDb\centralHub\Tests\Integration\ConfigStore
+ * @group   config-store
+ */
+class Tests_LoadConfigFromFilesystem extends Test_Case {
+
+	/**
+	 * Test loadConfigFromFilesystem() should merge defaults with config and return a store key.
+	 */
+	public function test_should_merge_defaults_and_return_store_key() {
+		$path_to_file  = CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		$defaults      = [
+			'aaa' => 37,
+			'eee' => 'Coding is fun!',
+		];
+		$merged_config = [
+			'aaa' => 'bbb',
+			'ccc' => 'ddd',
+			'eee' => 'Coding is fun!',
+		];
+
+		$this->assertSame( 'foo', loadConfigFromFilesystem( $path_to_file, $defaults ) );
+		$this->assertEquals( $merged_config, _the_store( 'foo' ) );
+	}
+
+	/**
+	 * Test loadConfigFromFilesystem() should store a config and return the store key.
+	 */
+	public function test_should_store_a_config_and_return_store_key() {
+		$path_to_file = CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		$config       = [
+			'aaa' => 'bbb',
+			'ccc' => 'ddd',
+		];
+
+		$this->assertSame( 'foo', loadConfigFromFilesystem( $path_to_file ) );
+		$this->assertSame( $config, _the_store( 'foo' ) );
+	}
+
+	/**
+	 * Test loadConfigFromFilesystem() should overwrite stored configuration and return store key from
+	 * file configuration.
+	 */
+	public function test_should_overwrite_store_and_return_store_key_from_file_config() {
+		$path_to_file = CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		$config       = [
+			'aaa' => 'bbb',
+			'ccc' => 'ddd',
+		];
+
+		$this->assertSame( 'foo', loadConfigFromFilesystem( $path_to_file ) );
+		$this->assertSame( 'foo', loadConfigFromFilesystem( $path_to_file ) );
+		$this->assertSame( $config, _the_store( 'foo' ) );
+	}
+
+	/**
+	 * Test loadConfigFromFilesystem() should overwrite stored configuration and return store key from
+	 * file configuration.
+	 */
+	public function test_should_throw_exception_when_no_store_key() {
+		$path_to_file = CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/config-with-params-only.php';
+		$this->expectException( \Exception::class );
+		$this->expectExceptionMessage(
+			sprintf( 'No store key exists in the %s configuration file.', $path_to_file )
+		);
+		Monkey\Functions\expect( '\KnowTheCode\ConfigStore\_merge_with_defaults' )->never();
+		Monkey\Functions\expect( '\KnowTheCode\ConfigStore\_the_store' )->never();
+
+		loadConfigFromFilesystem( $path_to_file );
```

Comment:
@rgadon107 Notice how this integration test is constructed.  What is it doing?

1. It's validating that `_load_config_from_filesystem()` does do what we expect:
      - Loads the config from file.
      - throws the error we expect
2. `_merge_with_defaults` does get called.
3. `_the_store ` does get called.

Writing a test in this way makes it very clear as to what behavior we expect and don't expect to happen.